sdk/messaging/azservicebus/example_receiver_test.go

```diff
@@ -86,3 +87,69 @@ func ExampleReceiver_ReceiveMessages() {
 		fmt.Printf("Received and completed the message\n")
 	}
 }
+
+func ExampleReceiver_DeadLetterMessage() {
+	// Create a sender client
+	sender, err := client.NewSender("myqueue", nil)
```

Comment:
I think we can clip the sending side of this out. It's mostly important to see the Receiver being created and then the dead letter action.